include/trng.h

```diff
@@ -0,0 +1,70 @@
+/***************************************************************************//**
+ *   @file   trng.h
+ *   @brief  Header file of true random number generator
+ *   @author Mihail Chindris (mihail.chindris@analog.com)
+********************************************************************************
+ *   @copyright
+ * Copyright 2020(c) Analog Devices, Inc.
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#ifndef TRNG_H
+#define TRNG_H
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <stdint.h>
+
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+
+/* TRNG Descriptor */
+struct rng_desc;
```

Comment:
should be `extern` in header
see this https://stackoverflow.com/questions/1433204/how-do-i-use-extern-to-share-variables-between-source-files

Reply:
struct rng_desc is a structure declaration. It is not a variable.

Reply:
oh, my bad
Then why isn't it properly declared here ? I see the actual declaration in `platform/aducm3029/trng.c`.

Reply:
Do you expect different trng structure forms based on platform ?

Reply:
Here is a declaration and in .c file is the definition. Yes, each platform can define its own structure this way.
In aducm are used only platform specific fields

Reply:
Ok, then two comments here:
- the buffer doesn't seem to be platform specific so you would have to duplicate it in every platform
- if you move the buffer in the generic one, you would need to to something similar to the `void *extra` thing that we do elsewhere

Reply:
The buffer is platform specific. It is the memory used by the BSP driver to hold the descriptor to don't allocate memory.
Inside adi_rng_Open is something like
`g_desc.dev = g_desc.dev_mem`